src/client/app/containers/editor.mode.ts

```diff
@@ -57,24 +60,37 @@ export class EditorMode {
 
     constructor(
         private _store: Store<fromRoot.State>,
-        private _effects: UIEffects
+        private _effects: UIEffects,
+        private _request: Request,
+        private _route: ActivatedRoute
     ) {
         this._store.select(fromRoot.getCurrent).subscribe(snippet => {
             this.isEmpty = snippet == null;
             this.snippet = snippet;
+
+            if (this.isEditorTryIt) {
+                window.parent.postMessage(this.snippet.id, environment.current.config.runnerUrl);
+                return;
+            }
         });
 
         this._store.select(fromRoot.getSharing).subscribe(sharing => {
             this.isDisabled = sharing;
         });
 
         this._store.dispatch(new GitHub.IsLoggedInAction());
+
+        this.checkForEditorTryIt();
     }
 
     get isAddinCommands() {
         return /commands=1/ig.test(location.search);
     }
 
+    get isEditorTryIt() {
+        return /#\/edit\//.test(location.hash)
```

Comment:
Can you get this from the route instead of depending on the hash test here?

Reply:
Added.